addons/web/static/tests/views/fields/statusbar_field_tests.js

```diff
@@ -623,4 +623,44 @@ QUnit.module("Fields", (hooks) => {
         await click(clickableButtons[clickableButtons.length - 1]);
         assert.verifySteps(["write"]);
     });
+
+    async function makeClickableReadonlyView(clickable, readonly) {
```

Comment:
we do not encourage this pattern. When you read a test, you prefer having everything there, instead of part of the test being dynamically generated in functions. We don't mind having a bit of duplicated code in tests.